src/rest-server/src/models/v2/group.js

```diff
@@ -412,15 +413,34 @@ const updateUserGroupAndVirtualCluster = async () => {
   }
 };
 
+const groupTypeVCCheck = async () => {
+  try {
+    const groupInfoList = await getAllGroup();
+    const vcList = await vcModel.prototype.getVcListAsyc();
+    for (let groupItem of groupInfoList) {
+      if (groupItem.extension.groupType && groupItem.extension.groupType === 'vc') {
+        if (!vcList || !(groupItem.groupname in vcList)) {
+          deleteGroup(groupItem.groupname);
+          logger.info(`Delete vc type group [${groupItem.groupname}]. Can't find it in yarn's vc list.`);
+        }
+      }
+    }
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log(error);
+  }
+};
+
 if (config.env !== 'test') {
   (async function() {
```

Comment:
why wrap it with async function

Reply:
All function is async, we should await it.

Reply:
however the groupTypeVCCheck is already async function, maybe you could use await directly

Reply:
https://stackoverflow.com/questions/39679505/using-await-outside-of-an-async-function